### ISSUE TITLE
MK priority list

### ISSUE THREAD

**Issue by polly64** [colouring-cities/colouring-london#682]
@matkoniecz this is our ongoing editable priority feature notepad. We use it to try and keep track of issues and tick them off when they're done, and we edit this first entry box rather than having a long thread as otherwise it gets very confusing.  Detailed discussions can then be undertaken under a separate issue number where required - we are also now putting an editable priority list at the top of each- so we as a team (and others) can quickly see components' progress status , and then any discussion can be included below this.

PRIORITY ISSUES
All relating to SECURITY/PRIVACY/INCLUSIVITY (extracted from https://github.com/colouring-london/colouring-london/issues/687)

- [x] **Add revised Data accuracy, Contributor Agreement, Code of contact pages** as https://github.com/colouring-london/colouring-london/issues/689
- [x] **Send Planning grant milestones and engineer spec to PH**
- [x] #696 Reactivate dark background 
- [x] Set dark background as default @matkoniecz 

### **For 'Current Use' category** 
- [ ] Change 'Unclassified presumed residential' on dropdown to 'Unclassified presumed non-residential'
- [ ] Add full title of 'Unclassified, presumed non-residential' to Land use key. can be on 2 lines
- [ ] Change text disclaimer to read 
- [ ] Suggest way of colouring a) Residential garden buildings' and b) Hotels a different colour form 'residential' and adding to the key.  We are now able to record these using the dropdowns butwe are still not able to visualise them as they are all showing as blue.
- [ ]  Suggest way of improving the 'copy' button. Not intuitive at moment- you have to click building, click land use, tick 'copy' and then also remember to click 'copy selected'. can we lose a step here?
- [ ] Using code from Age data section add a  source dropdown in the Current Use section to read as follows:
1. Expert/personal knowledge of building
2. Online streetview image
3. Planning authority data
4. Ordnance Survey data
5. Valuation Agency data
6. Other 
- [ ] Also add source link
- [ ] Also add a copy tool for the verification button - as you need to be able to quickly click on every building in a street where residential. If I have this and the colour change  I can actually run through them very quickly. Non domestic generally  has a very predictable spatial distribution along the pre 19th century routes.
- [ ] Also add a copy tool for  land use
- [ ] change key so you have - 
current blue = default setting
another blue or current blue with a border colour= verified residential

(To discuss - [ ] Change retail on key to  Retail/Professional services (to discuss)
Hotels also need to be coloured differently - even though these fall under residential in the NLUD classification system they don't work well here and should be  in commercial or recreation/ leisure. Not sure how best way to do this is- maybe to add a pink (recreation and leisure border and not include in the key? you can see it is a hotel when u click on the building, what do you think?)

For 'Age' Category

- [ ] Ask Maciek for file with my sources for age data for Camden
- [ ] Add sources to 'Age 'category. - Should you be able to add data without a source? discussion needed
- [ ] Move 'earliest' date box to above' latest' date box (Q)

For 'Community category'

- [ ] **Complete oustanding actions in  'Community category'** at https://github.com/colouring-london/colouring-london/issues/678

- [ ] **How can we quickly improve feedback mechanisms** 
a) Security button on interface 
b) Email re security issue
c) Email re recommendation
d) Shut down discussion forum as can't monitor and move to GitHub - discuss
e) Rename 'Privacy Policy' Menu page 'Privacy and Security' and add ref to information re  security procedures
- [ ] Exclude robots from https://colouring.london/leaderboard.html
- [ ] **Check with MZ all colour keys have been passed through checks for visually impaired**
- [ ] **How do we add organisation prefix on Leaderboard**
- [ ] **How do we check weblinks are ok?** 

OTHER ISSUES

In list below
Q= Assumed to be quick.  DN = Don't know how long this will take but not big, B= Big - but do of course adjust 
So for moment the list to chat with @mz8i about is as follows

- [ ]  Discuss idea of editing hierarchies?  e.g. for government datasets anyone can 'verify' but if they want to correct they need to go via government department moderators who then can accept/reject them and then contact us to update e.g. as already occurs for the database for statutorily protected buildings in England https://historicengland.org.uk/listing/the-list/minor-amendments/- [ ]  Activate code change  at https://github.com/colouring-london/colouring-london/issues/654 to create dark background (Q)
- [ ]  Complete a few outstanding issues recorded for Community section https://github.com/colouring-london/colouring-london/issues/678 (DN)
- [ ] Load in Ian Hall conservation data for London in Planning category to improve current dataset - link at http://www.bedfordpark.net/leo/planning/ (Q)
- [ ] Go through with PH where copy and verification tool needs to be added in all categories and add for all (Q)
- [ ] Implement extra copy and verification features as above (DN)

- [ ] Revist with PH and MZ all privacy and sign up pages and UPRN page descriptions to make them as clear to all users as possible. Use primary school teachers as an example of one of our target audiences to whom this information should be completely clear. (DN)

- [ ] Make last user in edit history more visible (Q)
- [ ] Go through all open issues with MZ and see if we can close any straight away (Q)
- [ ] Add icon for type of data capture/production method as suggested by MZ i.e for a) open bulk upload from specialist body, b) live streamed data, c) data crowdsourced at individual level , d) computationally generated using inference. This really would need to be applied to entries and their verification. Ideally we would be checking crowdsourced data using automated methods and vice versa. (DN)
- [ ] Talk to @mz8i  re moving wix pages over to main site and adjust to louis graphic design (about, categories, data ethics, Research programme, who's involved, plus text sheets for all 12 categories (Big piece of work) (B)
- [ ] Create a 'new item added' section/pop up/blog' i.e to tell people we have just expanded the 'Community' section
- [ ] Activate team section https://github.com/colouring-london/colouring-london/issues/677, and add in BREAM data (B)
- [ ] Create new Menu page called 'Animation and simulation links'  and incorporate PH text (Q)
- [ ] Activate Type - 'original use' - adding copy and verification buttons
and 'Architectural style' https://github.com/colouring-london/colouring-london/issues/679 (B)
- [ ] Add 'How to use video' (B)
- [ ] Build showcase section? or Turing REG? (B)
- [ ] Discuss updating of polygons with Tom and Maciek once we get access (from "Summary actions" 2021-11-12 - clarify)
- [ ] Follow up on security reporting once turing replies- if no replies Polly to contact Greg Mingas for thoughts.
- [ ] Talk to PH idea of map that will allow unchecked that have been added to be checked- once checked by moderating them will turn off colour so we always know what outstanding
- [ ] Check issue re backup further with Maciek and Tom - then check if how we back up our system is recorded anywhere on our site
- [ ] need to think about whether people join as organisations

**Comment by matkoniecz** [comment]
> How do we check weblinks are ok?

I see four ways to handle this:

- allow people to enter any links, monitor it and in case of malicious ones revert edit and ban people. Negative effect is that malicious links will be visible for some time, and importance of that depends on how often it happens and how important is to avoid it.
- allow people to enter any links, block known unwanted ones ( say, ones listed at https://github.com/StevenBlack/hosts#unified-hosts-file-with-base-extensions ). Negative effect: there are many unwanted sites not on such lists, new pornographic/scam/malicious sites are created daily - partially to avoid such lists.
- allow people to enter links leading only to websites on list of allowed sites. Negative effects: impossible to add safe sites that are not on the list, list needs to be maintained.
- allow people to enter any links, display only leading to websites on list of allowed sites. Negative effects: list needs to be maintained.

The questions are: 

Has this problem happened so far? Or is it treated as so significant that making contributing and site maintenance harder is a worthwhile tradeoff to close this potential attack?

How varied the sites used as sources are? How often it would be necessary to often add new ones? What about deployment in Athens? The likely would need to add many local sites  to list of valid ones.

In short: how significant effort is needed to maintain such list.

-----------------------------------------

Is blocking known unwanted sites sufficient? It will block most of attempts but someone malicious and motivated will succeed.

Or is it necessary to ensure that nothing will pass?

**Comment by matkoniecz** [comment]
Even with filtering domains it is still possible to add links like https://www.bbc.com/serious_insults_here

Note that sufficiently malicious people will manage to stuff offensive things in the year form (by setting year to 1488, see https://en.wikipedia.org/wiki/Fourteen_Words ) or by other content offensive by context.

So it is not possible to block malicious content completely and it is question how much effort should be put into that.

**Comment by polly64** [comment]
@matkoniecz could you add these comments to https://github.com/colouring-london/colouring-london/issues/687 too so we have them in the main privacy/security issue - thanks

**Comment by matkoniecz** [comment]
mirrored to https://github.com/colouring-london/colouring-london/issues/687#issuecomment-961259366

**Comment by polly64** [comment]
@matkoniecz to-do-list cleaned up

**Comment by matkoniecz** [comment]
superseded by other issues/discussions/plannings
